pkg/batch/executor.go

```diff
@@ -11,19 +11,31 @@ import (
 // dispatching new ones would start blocking.
 const RequestBufferSize = 1 << 17
 
+type Executable interface {
+	Execute() (interface{}, error)
+}
+
+type Batchable interface {
```

Comment:
Similarly, this should also be a `-er` if possible.  In this case the name also misled me, because in fact we do allow non-`Batchable` operations to be batched.